File: LeetCode Problems/1801 to 1900/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.cpp
class Solution {
public:
    int minPairSum(vector<int> &nums) {
        ios::sync_with_stdio(false);
        cin.tie(0);
        cout.tie(0);
        sort(nums.begin(), nums.end());
        int ans = INT_MIN;
        for (int l = 0, r = nums.size() - 1; l <= r; l++, r--) {
            ans = max(ans, nums[l] + nums[r]);
        }
        return ans;
    }
};

File: LeetCode Problems/1801 to 1900/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int ans = -1;
        for (int l = 0, r = nums.length - 1; l <= r; l++, r--) {
            ans = Math.max(ans, nums[l] + nums[r]);
        }
        return ans;
    }
}